src/__tests__/renderer.test.tsx

```diff
@@ -342,29 +365,23 @@ describe('slack jsx', () => {
       </ContextBlock>
     );
 
-    expect(message).toMatchSnapshot();
+    expect(await render(message)).toMatchSnapshot();
   });
 
-  //   // console.log(JSON.stringify(message, null, 2));
-  //   expect(message.blocks).toEqual([
-  //     {
-  //       elements: [
-  //         {
-  //           type: 'plain_text',
-  //           text: 'Hello, world',
-  //           emoji: true,
-  //         },
-  //         {
-  //           type: 'image',
-  //           image_url:
-  //             'https://api.slack.com/img/blocks/bkb_template_images/beagle.png',
-  //           alt_text: 'alt',
-  //         },
-  //       ],
-  //       type: 'context',
-  //     },
-  //   ]);
+  it('renders inline text elements with promises returning text elements', async () => {
+    const renderText = async () => {
+      const strings = await Promise.all(
```

Comment:
here is where the async `h` factory gets inconvenient. 

if you instead do 

```
const renderText = () => {
      const strings = ['foo', 'bar'].map(s => <Mention userId={s} />);
      return `world! ${strings}`;
    };

    const message = (
      <ContextBlock>
        <MarkdownText>Hello, {renderText()}</MarkdownText>
      </ContextBlock>
    );
```

you end up with 

```
"text": "Hello, world! [object Promise],[object Promise]",
```

because of this jsx->`h` replacement:

`yarn babel src/__tests__/renderer.test.tsx > out.js` :

```
    const renderText = () => {
      const strings = ['foo', 'bar'].map(s => _.slack.h(_.Mention, {
        userId: s`
      }));
      return `world! ${strings}`;
    };

    const message = _.slack.h(_.ContextBlock, null, _.slack.h(_.MarkdownText, null, "Hello, ", renderText()));
```

`_slack.h` returns a promise since it's an `async` function. and since it's immediately evaluated into the template string, as opposed to being queued into a tree of JSX nodes.

alternatively, for string-returning components, we could just make them function helpers. so `const mention = () => ''` and just use `mention(id)` instead of `<Mention id={id} />`.

thoughts, @davemeyer @shawnkoon ?

Reply:
hmm the function helper idea seems like it would be confusing from a usage perspective. but not sure of a better idea currently. will think on it

Reply:
I think this is good as it shows intention behind the test :+1:, maybe creating **async** string-returning components helper method could make test less complex/more readable if that is the concern here.

Just to validate, when you've mentioned:
```js
const strings = ['foo', 'bar'].map(s => <Mention userId={s} />);
// returns [object Promise],[object Promise]",
```

You mean `.map(s => <Mention userId={s} />);` right?

Reply:
well in my example above, @shawnkoon what i meant is that if i leave `render` OUT, you get the unresolved `[object Promise]` in the output. so you NEED render for components in inline template strings

Reply:
so yeah i kinda agree, @davemeyer let's leave it as JSX and add some documentation around it?

Reply:
> well in my example above, @shawnkoon what i meant is that if i leave `render` OUT, you get the unresolved `[object Promise]` in the output. so you NEED render for components in inline template strings

whaaa that's bit unexpected.

Reply:
yeah, agreed it is "unexpected". but feels less unexpected than using helpers in jsx. also, for usage example @shawnkoon https://github.com/asynchronous-dev/botany/pull/582/files?file-filters%5B%5D=.graphql&file-filters%5B%5D=.json&file-filters%5B%5D=.lock&file-filters%5B%5D=.tsx&file-filters%5B%5D=.yaml#diff-a9e314708cfcfb3eb2c31f265513591aR75

and agreed @andycmaj , this isn't a huge change assuming the documentation explains it.

Reply:
yeah exactly... but basically the cost of being able to use an async factory function